pkg/rtc/dynacastmanager.go

```diff
@@ -0,0 +1,265 @@
+package rtc
+
+import (
+	"sync"
+	"time"
+
+	"github.com/bep/debounce"
+	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
+
+	"github.com/livekit/livekit-server/pkg/rtc/types"
+	"github.com/livekit/livekit-server/pkg/utils"
+)
+
+type DynacastManagerParams struct {
+	DynacastPauseDelay time.Duration
+	Logger             logger.Logger
+}
+
+type DynacastManager struct {
```

Comment:
😍 this is fantastic!  do you think we should move the test cases over from MediaTrackTest to here too?

Reply:
Will move them.